superchain/internal/codegen/main.go

```diff
@@ -100,21 +100,41 @@ func main() {
 	fmt.Println("Wrote chainList.toml file")
 
 	// Write all chain configs to a single file
-	type Configs struct {
-		Configs []ChainConfig `toml:"configs"`
+	type Superchain struct {
+    Name string `toml:"name"`
+    Config SuperchainConfig `toml:"superchain"`
+		ChainConfigs []ChainConfig `toml:"configs"`
 	}
-	cfgs := make([]ChainConfig, 0)
-	for _, chain := range OPChains {
-		cfgs = append(cfgs, *chain)
-	}
-	sort.Slice(cfgs, func(i, j int) bool {
-		return cfgs[i].ChainID < cfgs[j].ChainID
-	})
-	if len(cfgs) != 0 {
-		var buf bytes.Buffer
-		if err := toml.NewEncoder(&buf).Encode(Configs{Configs: cfgs}); err != nil {
-			panic(fmt.Errorf("failed to marshal toml: %w", err))
-		}
+  superchains := make([]Superchain, 0)
+  for _, sc := range Superchains {
+    chainConfigs := make([]ChainConfig, 0)
+    for _, chainId := range sc.ChainIDs {
+      chain := OPChains[uint64(chainId)]
+      if chain == nil {
```

Comment:
Is it more idiomatic to do
```suggestion
      chain, found := OPChains[uint64(chainId)]
      if !found {
```